src/llmcompressor/modifiers/quantization/awq/base.py

```diff
@@ -0,0 +1,467 @@
+        logger.info(
+            f"Running {class_name} calibration with "
+            f"{len(calibration_dataloader)} samples..."
+        )
+        if not calibration_dataloader:
+            raise ValueError(
+                "Calibration data loader not set, must populate the calib_data field of"
+                " CompressionSession to run the AWQ modifier"
+            )
+
+        run_calibration_forward(
+            model,
+            calibration_dataloader,
+            self.num_calibration_steps,
+            self.calibration_function,
+        )
+
+        # remove the hooks now that we are done calibrating
+        for hook in self.hooks_:
+            hook.remove()
+        del self.hooks_
+
+    def _concat_collected_activations(self):
+        """
+        Concatenate the collected activation values from each forward pass into a single
+        tensor for each layer
+
+        :postcondition: each layer in self.scales_ will have a single tensor containing
+            all the activation values seen during calibration
+        """
+        for mapping in self.resolved_mappings_:
+            name = mapping.smooth_name
+            self.scales_[name].inps = torch.cat(self.scales_[name].inps, dim=0)
+
+        torch.cuda.empty_cache()
+
+    @torch.no_grad()
+    def _apply_smoothing(self, model: Module):
+        """
+        Calculate the best scaling factors for each layer to smooth activations and
+        apply the scaling factors to the weights of the next layer to offset the
+        smoothing
+
+        :param model: model to apply smoothing to
+        """
+        logger.info("Smoothing activation scales...")
+        for mapping in self.resolved_mappings_:
+            smooth_layer = mapping.smooth_layer
+            balance_layers = mapping.balance_layers
+
+            activations = self.scales_[mapping.smooth_name].inps
+            module2inspect = smooth_layer
+
+            # [STEP 1]: Compute per-channel mean of normalised weights
+            # All layer weights are concatted together
+            weight = torch.cat([_m.weight for _m in smooth_layer], dim=0)
+            org_shape = weight.shape
+            # The weights are reshaped to be organised by quantization group
+            weight = weight.view(-1, self.group_size)
+            # Calculates the relative magnitude of the weights within
+            # each of the quantization groups, and rescales each group
+            # individually so that each group has weights on a 0-1 scale.
+            w_scale = weight.abs() / (weight.abs().amax(dim=1, keepdim=True) + 1e-6)
+            # Resizes the rescaled weight matrix back up to its original dimensions
+            w_scale = w_scale.view(org_shape)
+            # Gets the average rescaled magnitude for each output channel
+            w_mean = w_scale.mean(0)
+
+            # [STEP 2]: Compute per-channel mean of the input activation with chunking
+            # move inp to cpu to avoid memory leak
```

Comment:
What operation causes the memory leak? Should the tensor be moved back to gpu after?